dashboard-api/internal/types/studyconf.go

```diff
@@ -172,6 +177,88 @@ type RecruitmentConf struct {
 	MaxSamplePerArm    int    `json:"max_sample_per_arm,omitempty"`
 }
 
+// DestinationConf as we have multiple types of destination
+// we use a blank interface.
+// Destination is where to send a recruitee to
+type Destination interface {
+	SetName(n string)
+}
+
+type DestinationConf []Destination
+
+// UnmarshalJSON is a custom unmarshaller to handle the different types
+// of destinations
+func (d *DestinationConf) UnmarshalJSON(data []byte) error {
+	var jsonDest []struct {
+		Name string `json:"name"`
+		FlyDestination
+		WebDestination
+		AppDestination
+	}
+
+	if err := json.Unmarshal(data, &jsonDest); err != nil {
+		return err
+	}
+
+	dests := []Destination{}
+
+	for _, jd := range jsonDest {
+		switch {
+		//WebDestination should always have URLTemplate
+		//set
+		case jd.URLTemplate != "":
```

Comment:
Interesting. Duck typing in Go. I like it. 

But I think after you've ducktyped, determined what it is, you should just re-marshal the JSON into the correct type, no? 

Another option -> use `gjson` to avoid marshaling until you know what it is.

Reply:
This smells funky to me... I can't really understand what's going on, tbh. 

It feels like we're marshalling it into a Go Type, but then putting it into a weird interface that's anyways useless, so we're throwing away all type information. At that point, what's the point? Why not just leave it as rawJson?

Reply:
So this structure allows for us to have a list of different types (This is the object of the interface), so when we add the validation we can specify required fields etc based on the types (The types are still there). I agree with the GJson but keeping them in raw JSON will make it hard for validation top occur.